TickSpec/ScenarioGen.fs

```diff
@@ -392,29 +392,30 @@ let defineStepMethod
         emitArray gen providerField parsers t x
     )
     // Emit table argument
-    line.Table |> Option.iter (fun table ->
-        let p = ps.[ps.Length-1]
-        let t = p.ParameterType
-        if t = typeof<Table> then emitTable gen table
-        elif t.IsArray then
-            let found =
-                t.GetElementType().GetConstructors()
-                |> Seq.tryFind (fun c -> c.GetParameters().Length = table.Header.Length)
-            match found with
-            | Some ci -> emitObjectArray gen providerField parsers t table ci
-            | None ->
+    if ps.Length > args.Length then
```

Comment:
Do you want to have an else with an exception - similarly as you have in ScenarioRun?

Reply:
In this case I don't think it is necessary. It appears that additional capture arguments are silently dropped when there are not enough parameters declared, so doing the same for Tables and Lists feels consistent here.

Reply:
However, in ScenarioRun you are throwing an exception in such case. Do I understand it correctly?

Reply:
Yes there are two different styles being used.
In `ScenarioRun.invokeStep` array indexing is used to access the parameters, which may throw an exception if too few parameters are defined on the method, eg: 

```fsharp
    let ps = meth.GetParameters()
    let buildArgs (xs:string[]) =
        args |> Array.mapi (fun i s ->
            let p = ps.[i].ParameterType // <--- exception thrown if too few parameters
```

vs `ScenarioGen.defineStepMethod` which uses `Seq.zip` to match the arguments with the method parameters.

```fsharp
    // Emit arguments
    let ps = mi.GetParameters()
    let zipped = Seq.zip args ps // <--- additional arguments dropped if too few parameters
```